fedora_messaging/tests/unit/test_session.py

```diff
@@ -195,6 +195,27 @@ def test_publish_reconnect_failed(self):
         self.assertEqual(self.publisher._connection, connection_mock)
         connection_mock.close.assert_called_once()
 
+    def test_publish_reconnect_failed_ChannelClosed(self):
+        # The publisher must try to re-establish a connection on publish, and
+        # close the connection if it can't be established.
+        self.publisher._channel.publish.side_effect = pika_errs.ConnectionClosed(
+            200, "I wanted to"
+        )
+        connection_class_mock = mock.Mock()
+        connection_mock = mock.Mock()
+        connection_class_mock.return_value = connection_mock
+        connection_mock.channel.side_effect = pika_errs.ChannelClosed(
+            404, "NOT_FOUND - no exchange 'amq.fake' in vhost '/'"
+        )
```

Comment:
Mock objects return Mock objects by default, so I *believe* you can rewrite this as:

```
connection_class_mock = mock.Mock()
connection_class_mock.return_value.channel.side_effect = pika_errs.ChannelClosed(...)
```

It makes no functional change and it's a copy of existing code, but just an FYI :cake: